cmd/kube-proxy/app/server_others.go

```diff
@@ -76,22 +75,46 @@ func (o *Options) platformApplyDefaults(config *proxyconfigapi.KubeProxyConfigur
 	klog.V(2).InfoS("DetectLocalMode", "localMode", string(config.DetectLocalMode))
 }
 
-// createProxier creates the proxy.Provider
-func (s *ProxyServer) createProxier(config *proxyconfigapi.KubeProxyConfiguration) (proxy.Provider, error) {
-	var proxier proxy.Provider
-	var err error
+func (s *ProxyServer) platformCheckSupported() (ipv4Supported, ipv6Supported, dualStackSupported bool, err error) {
+	execer := exec.New()
+	ipt := utiliptables.New(execer, utiliptables.ProtocolIPv4)
+	ipv4Supported = ipt.Present()
+	ipt = utiliptables.New(execer, utiliptables.ProtocolIPv6)
+	ipv6Supported = ipt.Present()
+
+	// The Linux proxies can always support dual-stack if they can support both IPv4
+	// and IPv6.
+	dualStackSupported = ipv4Supported && ipv6Supported
+
+	if !ipv4Supported && !ipv6Supported {
+		err = fmt.Errorf("iptables is not available on this host")
+	} else if !ipv4Supported {
+		klog.InfoS("No iptables support for family", "ipFamily", v1.IPv4Protocol)
+	} else if !ipv6Supported {
+		klog.InfoS("No iptables support for family", "ipFamily", v1.IPv6Protocol)
+	}
 
-	var nodeInfo *v1.Node
-	if config.DetectLocalMode == proxyconfigapi.LocalModeNodeCIDR {
+	// LocalModeNodeCIDR is only supported if node.Spec.PodCIDRs is being set. (Note:
```

Comment:
This is a little weird in terms of surprising side-effects.  What if we factored it into a `s.prepLocalMode()` or something and just call it explicitly in the same area that `platformCheckSupported()` is called?

Reply:
ok, there's already `s.platformSetup()`... it's currently called _after_ the Proxier is created, but for both linux and windows it doesn't do anything that couldn't just as easily be done _before_, so I just moved that and added the localdetector stuff to that.